internal/service/connect/instance_storage_config_data_source_test.go

```diff
@@ -0,0 +1,377 @@
+				),
+			},
+		},
+	})
+}
+
+func testAccInstanceStorageConfigDataSource_S3Config(t *testing.T) {
+	rName := sdkacctest.RandomWithPrefix("resource-test-terraform")
+	rName2 := sdkacctest.RandomWithPrefix("resource-test-terraform")
+	resourceName := "aws_connect_instance_storage_config.test"
+	datasourceName := "data.aws_connect_instance_storage_config.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, connect.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		Steps: []resource.TestStep{
+			{ // nosemgrep:ci.test-config-funcs-correct-form
+				Config: testAccInstanceStorageConfigDataSourceConfig_S3Config(rName, rName2),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "association_id", resourceName, "association_id"),
+					resource.TestCheckResourceAttrPair(datasourceName, "instance_id", resourceName, "instance_id"),
+					resource.TestCheckResourceAttrPair(datasourceName, "resource_type", resourceName, "resource_type"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.#", resourceName, "storage_config.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.#", resourceName, "storage_config.0.s3_config.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.0.bucket_name", resourceName, "storage_config.0.s3_config.0.bucket_name"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.0.bucket_prefix", resourceName, "storage_config.0.s3_config.0.bucket_prefix"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.0.encryption_config.#", resourceName, "storage_config.0.s3_config.0.encryption_config.#"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.0.encryption_config.0.encryption_type", resourceName, "storage_config.0.s3_config.0.encryption_config.0.encryption_type"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.s3_config.0.encryption_config.0.key_id", resourceName, "storage_config.0.s3_config.0.encryption_config.0.key_id"),
+					resource.TestCheckResourceAttrPair(datasourceName, "storage_config.0.storage_type", resourceName, "storage_config.0.storage_type"),
+				),
+			},
+		},
+	})
+}
+
+func testAccInstanceStorageConfigDataSourceConfig_base(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_connect_instance" "test" {
+  identity_management_type = "CONNECT_MANAGED"
+  inbound_calls_enabled    = true
+  instance_alias           = %[1]q
+  outbound_calls_enabled   = true
+}
+`, rName)
+}
+
+func testAccInstanceStorageConfigDataSourceConfig_kinesisFirehoseConfig(rName, rName2, rName3, rName4, rName5 string) string {
```

Comment:
Can the same `rName` be used for all resources? That would mean less extra setup for the test. e.g.,

```suggestion
func testAccInstanceStorageConfigDataSourceConfig_kinesisFirehoseConfig(rName string) string {
```